src/txmempool.h

```diff
@@ -34,6 +34,16 @@
 
 class CChain;
 
+#ifndef BITCOIN_MEMPOOL_SET_DEFINITIONS_H
```

Comment:
1. I don't think this guard is necessary.
2. `DisconnectedTransactionsIteratorImpl` and `IndexedDisconnectedTransactionsImpl` don't seem to exist
3. Need to remove `mempool_set_definitions.h` to _actually_ get boost out of the headers, no?

Reply:
Will fixup points 2 and 3, must have slipped through the rebase.

> I don't think this guard is necessary.

The reason I added this here is that if there is an implementation that first includes the `mempool_set_definitions` and then the `txmempool` we don't redefine these data structures. Is there a better way to do this?

Reply:
Correct me if I'm misunderstanding, but I believe this is the pattern?
```c++
struct foo;
struct foo
{
    int bar;
};
struct foo;
int main(){}
```

If so, I (and gcc/clang :) don't see an issue with that.

Reply:
(Didn't mean to get in the weeds with review, I realize you're still looking for concept ACKs. The `mempool_set_definitions.h` thing is all I really care about at this stage)

Reply:
> If so, I (and gcc/clang :) don't see an issue with that.

I thought for some reason that this was discouraged, but you're obviously right.

> Didn't mean to get in the weeds with review,

np, any engagement here is much appreciated :)